### PR TITLE
rm symlink for terminal use, recommend .app

### PR DESCRIPTION
**Pull request by devinrhode2**
Also added a link more tips for Emacs on MacOS on the EmacsWiki

### CODE REVIEW

**Comment by railwaycat** [comment]
Thanks @devinrhode2 ! 

But I think there may still some people (yes, include me...) using this emacs-mac not only from GUI but also in terminal. Because the one shipped by Apple is quite old and almost totally not usable, especially for someone dealing with characters more than ASCII. 

I am not quite sure about the EmacsWiki link, seems it does the following two things: 1. start Emacs from a certain directory; 2. start Emacs.app with system PATH environment. 

For the first thing, isn't it better to set it in `init.el` file? like
```
  (setq default-directory "~/")
```
For the second one,  I'm not sure if the official NS port still have this issue, but Mac port fixed it long time ago:) by `Emacs.app/Contents/MacOS/Emacs.sh`.

**Comment by devinrhode2** [comment]
Yeah.. because eshell isn't a complete/real terminal emulator yet, there is definitely the case for using emacs in the terminal (it's convenient for a variety of already defined workflows) - since the one by apple is almost totally unusable, can we recommend a terminal (or two) for using Emacs Mac Port from the command line?

I would imagine we could also safely recommend zsh if there was any reason too, but I'm not sure we'd want to go as far as to recommend the [oh-my-zsh emacs plugin](https://github.com/robbyrussell/oh-my-zsh/blob/master/plugins/emacs/emacs.plugin.zsh) yet. This plugin uses emacsclient for launching editing buffers, which seems pretty ideal.

TLDR; It would be awesome to at least attempt to recommend a basic terminal-editor workflow.

Have you read "Mastering Eshell"? I need to go through that to figure out if eshell can effectively handle 98% of my terminal use. I know it should be able to handle the majority of my terminal use, but I'm not sure it will be able to handle 98% of it.

I wish emacs on mac+linux would just implement a full terminal emulator to **completely** handle your terminal needs. I just wish there was an integrated terminal solution for emacs that took care of __all your needs__ (except for rare edge cases).

TLDR; It would be awesome if there was an undeniably best way to run terminal commands with emacs.

Regarding the EmacsWiki link:
1. I would imagine it's better to set that line in your **`init.el`**, it seems more straightforward, like a better way to do this, and the init.el can be version controlled. The other way recommended in the EmacsWiki is pretty convoluted.
2. Emacs.sh: I had no idea this script existed, should we point to that instead of this gist: https://gist.github.com/4043945?
